crates/katana/executor/src/implementation/blockifier/mod.rs

```diff
@@ -98,8 +98,10 @@ impl<'a> StarknetVMProcessor<'a> {
 
         // TODO: should we enforce the gas price to not be 0,
         // as there's a flag to disable gas uasge instead?
-        let eth_l1_gas_price = unsafe { NonZeroU128::new_unchecked(header.gas_prices.eth) };
-        let strk_l1_gas_price = unsafe { NonZeroU128::new_unchecked(header.gas_prices.strk) };
```

Comment:
i didnt think much when adding this just to get things working initially and was planning on removing the unsafe later on, but never did. So if removing the `new_unchecked` here would solve the problem, then lets just fix this instead.